### PR TITLE
recursive directory scan for source-filesystem

### PR DESCRIPTION
**Pull request by odama626**
fixes https://github.com/tonyketcham/flatbread/issues/4

### CODE REVIEW

File: packages/source-filesystem/src/utils/getValidNodesFilenames.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        )
+    ).then((nodes) => nodes.flat());
+  }
+
+  nodes = nodes.map((node) => {
+    if (leaf) {
+      node.data[leaf.name] = getSegmentData(node, leaf);
+    }
+    return node;
+  });
```
Comment (tonyketcham) [inline packages/source-filesystem/src/utils/getValidNodesFilenames.ts:90]: Might be good to sprinkle in some inline comments here to briefly explain the steps as this took a few cognitive passes to understand

**Comment by tonyketcham** [comment]
We should also add a blurb on this feature to the `source-filesystem` README 😁

File: packages/source-filesystem/src/utils/getValidNodesFilenames.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+ *
+ * @param path The directory to read from
+ * @param extensions File extensions to parse
+ * @returns A list of content nodes' filenames
+ */
+export default async function getValidNodesFilenames(
+  path: string,
+  extensions?: string[],
+  { readDirectory = readDir } = {}
+): Promise<FileNode[]> {
```
[the remark this answers is not in the document]
Comment (odama626) [reply packages/source-filesystem/src/utils/getValidNodesFilenames.ts:50]: what about gatherFileNodes?

**Review by tonyketcham — APPROVED** [review]
LGTM! Thanks for taking the time to make those changes 👏

**Pull request merged**